flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -149,11 +149,14 @@
 @Public
 public class StreamExecutionEnvironment implements AutoCloseable {
 
-    private static final List<CollectResultIterator<?>> collectIterators = new ArrayList<>();
+    private static final List<CollectResultIterator<?>> collectIterators =
```

Comment:
Why do we need a `static` variable here? Let's say there are two instances of `StreamExecutionEnvironment`. Each environment should only care about its own job. But in `executeAsync` iterators for other jobs might be affected too.

@zentol please take a look.

Reply:
I agree that the variable does not need to be static since it is accessed only by current environment.

Reply:
I think it's just an accident that it is static.

Reply:
@zentol Thank you for your clarification, I will remove the static modifier for this field.